### PR TITLE
Resolving issue #12 for locating the application under Mac with JDK 8u40...

### PR DESCRIPTION
**Pull request by Maxoudela**
....

### CODE REVIEW

File: api/src/main/java/com/vinumeris/updatefx/UpdateFX.java [hunk 1]
```diff
@@ -85,9 +85,17 @@ public static Path findAppExecutable(Path appInstallDir) throws IOException {
         if (os.contains("mac")) {
             // Returns path to the .app directory, which normally contains only a Contents directory.
             List<Path> exes = Utils.listDir(appInstallDir.resolve("Contents/MacOS"));
-            if (exes.size() != 1)
-                throw new IllegalStateException("Found unknown number of app executables");
-            return exes.get(0);
+            for (Path exe : exes) {
+                /**
+                 * Since 8u40, the packager introduced a "libpackager.dylib"
```
Comment (mikehearn) [inline api/src/main/java/com/vinumeris/updatefx/UpdateFX.java]: This is probably going to exist on other platforms too. So perhaps we should rather go with your original suggestion of ignoring any file with an extension, or perhaps matching on the "libpackager" name and ignoring the extension (which could be dll, .so etc)

File: api/src/main/java/com/vinumeris/updatefx/UpdateFX.java [hunk 2]
```diff
@@ -85,9 +85,17 @@ public static Path findAppExecutable(Path appInstallDir) throws IOException {
...
+                 * next to the application. So we only need to ignore that file
+                 * and take the other one.
+                 */
+                if (!exe.toString().toLowerCase().endsWith(".dylib")) {
+                    return exe;
+                }
+            }
+            throw new IllegalStateException("We did not found the application.");
```
Comment (mikehearn) [inline api/src/main/java/com/vinumeris/updatefx/UpdateFX.java]: Typo: we did not FIND

Comment (Maxoudela) [inline api/src/main/java/com/vinumeris/updatefx/UpdateFX.java]: Oups sorry for my bad english

File: api/src/main/java/com/vinumeris/updatefx/UpdateFX.java (hunk 1 above)
Comment (Maxoudela) [inline api/src/main/java/com/vinumeris/updatefx/UpdateFX.java]: I don't understand. We are talking about the mac case here right?
You're suggesting I only take the file that do not have extension ?

Comment (mikehearn) [inline api/src/main/java/com/vinumeris/updatefx/UpdateFX.java]: Oh sorry, you are right, this is my mistake. I forgot that this code is inside an if (mac) {}

Yes, now this looks good to me. If you fix the typo then I'll merge. Thanks!

Comment (Maxoudela) [inline api/src/main/java/com/vinumeris/updatefx/UpdateFX.java]: I have another version where I check if the file name has a "." (dot) inside. If not, then I take the file. Thus we avoid putting any string directly for comparison, does that sound better?

if (!exe.getFileName().toString().contains(".")) {
                    return exe;
                }

Comment (mikehearn) [inline api/src/main/java/com/vinumeris/updatefx/UpdateFX.java]: No sorry, you were right and I was wrong. Apologies for the confusion. The code as you submitted it is fine.

**Comment by mikehearn** [comment]
Thanks for your contribution! If it's OK by you, I will add you to an AUTHORS file so you get credit for taking part. Is it OK to add your email address there too?

**Comment by Maxoudela** [comment]
Yep feel free to add it!

**Pull request merged**